smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -232,48 +242,29 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
 
         // ensure the networkDescriptor matches
         if (!claimData.doublePeg) {
-            require(verifyNetworkDescriptor(claimData.networkDescriptor), "INV_NET_DESC");
+            require(_verifyNetworkDescriptor(claimData.networkDescriptor), "INV_NET_DESC");
         }
         
-        // ensure there are no duplicate signers
-        require(
-            !findDup(signatureData), "DUP_SIGNER"
-        );
+        (bool dup, uint256 pow) = getSignedPowerAndFindDup(signatureData, hashDigest);
+        require(!dup, "DUP_SIGNER");
 
-        // update the nonce
-        lastNonceSubmitted = claimData.nonce;
+        require(getProphecyStatus(pow), "INV_POW");
 
-        // check that all signers are validators and are unique
-        for (uint256 i = 0; i < signatureData.length; i++) {
-            require(isActiveValidator(signatureData[i].signer), "INV_SIGNER");
-            require(
-                verifySignature(
-                    signatureData[i].signer,
-                    hashDigest,
-                    signatureData[i]._v,
-                    signatureData[i]._r,
-                    signatureData[i]._s
-                ) == true,
-                "INV_SIG"
-            );
-        }
-
-        uint256 signedPower = getSignedPower(signatureData);
-        require(getProphecyStatus(signedPower), "INV_POW");
-
-        // if we are double pegging AND we don't control the token, we deploy a new smart contract
         address tokenAddress;
-        if(claimData.doublePeg && !_isManagedToken(claimData.tokenAddress)) {
-            tokenAddress = _createNewBridgeToken(
-                claimData.tokenSymbol,
-                claimData.tokenName,
-                claimData.tokenAddress,
-                claimData.tokenDecimals,
-                claimData.networkDescriptor
-            );
-        } else if(claimData.doublePeg) {
-            // if we are double pegging and already control the token, then we are going to need to get the address on this chain
-            tokenAddress = sourceAddressToDestinationAddress[claimData.tokenAddress];
+        if(claimData.doublePeg) {
+            if(!_isManagedToken(claimData.tokenAddress)) {
+              // if we are double pegging AND we don't control the token, we deploy a new smart contract
+              tokenAddress = _createNewBridgeToken(
+                  claimData.tokenSymbol,
+                  claimData.tokenName,
+                  claimData.tokenAddress,
+                  claimData.tokenDecimals,
+                  claimData.networkDescriptor
```

Comment:
Let's add a new field `cosmosDenom` that will be a part of all bridgetokens.

There could be downstream implications on pre-existing bridgetokens from doing this.

Reply:
As we talked about on our call, I'll create a new branch to implement the `cosmosDenom`